crates/router/src/core/payments/routing.rs

```diff
@@ -2052,88 +2043,91 @@ pub async fn perform_success_based_routing(
-                        score: label_with_score.score,
-                    },
+        let closure = || async {
+            let success_based_connectors_result = client
+                .calculate_success_rate(
+                    profile_id.get_string_repr().into(),
+                    success_based_routing_configs,
+                    success_based_routing_config_params,
+                    routable_connectors,
+                    state.get_grpc_headers(),
                 )
-                .collect(),
-            routing_approach: match success_based_connectors.routing_approach {
-                0 => api_routing::RoutingApproach::Exploration,
-                1 => api_routing::RoutingApproach::Exploitation,
-                _ => {
-                    return Err(errors::RoutingError::GenericNotFoundError {
-                        field: "routing_approach".to_string(),
-                    })
-                    .change_context(errors::RoutingError::GenericNotFoundError {
-                        field: "unknown routing approach from dynamic routing service".to_string(),
-                    })
-                    .attach_printable("unknown routing approach from dynamic routing service")
+                .await
+                .change_context(errors::RoutingError::SuccessRateCalculationError)
+                .attach_printable(
+                    "unable to calculate/fetch success rate from dynamic routing service",
+                );
+
+            match success_based_connectors_result {
+                Ok(success_response) => {
+                    let updated_resp = utils::CalSuccessRateEventResponse::try_from(
+                        &success_response,
+                    )
+                    .change_context(errors::RoutingError::SuccessBasedRoutingConfigError)
+                    .attach_printable(
+                        "unable to convert SuccessBasedConnectors to CalSuccessRateEventResponse",
+                    )?;
+
+                    Ok(Some(updated_resp))
                 }
-            },
+                Err(e) => {
+                    logger::error!(
+                        "unable to calculate/fetch success rate from dynamic routing service: {:?}",
+                        e.current_context()
+                    );
+
+                    Err(error_stack::report!(
+                        errors::RoutingError::SuccessRateCalculationError
+                    ))
+                }
+            }
         };
 
-        routing_event.set_response_body(&event_response);
-        routing_event.set_routing_approach(event_response.routing_approach.to_string());
+        let events_response = utils::routing_events_wrap(
+            state,
+            routing_events_wrapper,
+            closure,
+            "SuccessRateCalculator.FetchSuccessRate".to_string(),
+            RoutingEngine::IntelligentRouter,
+            ApiMethod::Grpc,
+        )
+        .await?;
```

Comment:
```suggestion
        let events_response = routing_events_wrapper.trigger_event(
            state,
            closure,
            "SuccessRateCalculator.FetchSuccessRate".to_string(),
            RoutingEngine::IntelligentRouter,
            ApiMethod::Grpc,
        )
        .await?;
```

move routing_events_wrap to RoutingEventsWrapper impl and rename it to trigger_event